homeassistant/components/isy994/services.py

```diff
@@ -0,0 +1,408 @@
+        value = service.data.get(CONF_VALUE)
+        init = service.data.get(CONF_INIT, False)
+        isy_name = service.data.get(CONF_ISY)
+
+        for config_entry_id in hass.data[DOMAIN]:
+            isy = hass.data[DOMAIN][config_entry_id][ISY994_ISY]
+            if isy_name and not isy_name == isy.configuration["name"]:
+                continue
+            variable = None
+            if name:
+                variable = isy.variables.get_by_name(name)
+            if address and vtype:
+                variable = isy.variables.vobjs[vtype].get(address)
+            if variable is not None:
+                hass.async_add_executor_job(variable.set_value, value, init)
+                return
+        _LOGGER.error("Could not set variable value. Not found or enabled on the ISY.")
+
+    async def async_cleanup_registry_entries(service) -> None:
+        """Remove extra entities that are no longer part of the integration."""
+        entity_registry = await er.async_get_registry(hass)
+        config_ids = []
+        current_unique_ids = []
+
+        for config_entry_id in hass.data[DOMAIN]:
+            config_entities = er.async_entries_for_config_entry(
+                entity_registry, config_entry_id
+            )
+            config_ids.extend(
+                [
+                    (item.unique_id, item.entity_id, item.device_id)
+                    for item in config_entities
+                ]
+            )
+
+            hass_isy_data = hass.data[DOMAIN][config_entry_id]
+            uuid = hass_isy_data[ISY994_ISY].configuration["uuid"]
+
+            for platform in SUPPORTED_PLATFORMS:
+                for node in hass_isy_data[ISY994_NODES][platform]:
+                    if hasattr(node, "address"):
+                        current_unique_ids.append(f"{uuid}_{node.address}")
+
+            for platform in SUPPORTED_PROGRAM_PLATFORMS:
+                for _, node, _ in hass_isy_data[ISY994_PROGRAMS][platform]:
+                    if hasattr(node, "address"):
+                        current_unique_ids.append(f"{uuid}_{node.address}")
+
+            for node in hass_isy_data[ISY994_VARIABLES]:
+                if hasattr(node, "address"):
+                    current_unique_ids.append(f"{uuid}_{node.address}")
+
+        extra_entities = [
+            (entity_id, device_id)
+            for unique_id, entity_id, device_id in config_ids
+            if unique_id not in current_unique_ids
+        ]
+
+        for entity_id, _ in extra_entities:
+            if entity_registry.async_is_registered(entity_id):
+                entity_registry.async_remove(entity_id)
+
+        _LOGGER.info(
+            "Cleaning up ISY994 Entities and devices: Config Entries: %s, Current Entries: %s, "
+            "Extra Entries Removed: %s",
+            len(config_ids),
+            len(current_unique_ids),
+            len(extra_entities),
+        )
+
+    async def async_reload_config_entries(service) -> None:
```

Comment:
Why do we need a service for this?

Reply:
New devices, programs, etc. that are linked or updated in the ISY are not automatically discovered by the PyISY module without restarting (https://github.com/automicus/PyISY/issues/69). This service forces an unload and reload to allow these new or changed entries to be picked up and sorted as entities inside Home Assistant without having to restart the entire instance or delete and re-add the config.

Reply:
Ok. It would be nice if the library would handle this in the future, but it's ok to add a service for now.

Reply:
Roger that. Started laying some ground work for it in current PRs over there, so hopefully it's not too far off.